assets/js/components/dialog.js

```diff
@@ -89,7 +94,7 @@ class Dialog extends Component {
 										{ subtitle }
 									</p>
 								}
-								<section id="remove-module-dialog-description" className="mdc-dialog__content">
+								<section id={ describedByID } className="mdc-dialog__content">
```

Comment:
Let's do the `provides && provides.length` check here and not even render the section element otherwise. Currently, it could be a section with an empty list that is not needed.